staticx/api.py

```diff
@@ -71,13 +73,29 @@ def _cleanup(self):
             self.sxar.close()
             self.sxar = None
 
+    def _rewrite_tmpdir(self, bootloader):
+        if self.tmprootdir:
+            max_len = len(self.bootloader_repl_str) + 1024
+            if len(self.tmprootdir) > max_len:
+                raise InvalidInputError("tmprootdir needs to have maximum "+str(max_len)+" characters")
+
+            f = open(bootloader, 'rb')
+            repl = self.tmprootdir
+            repl_str = self.bootloader_repl_str + (" " * self.bootloader_tmpdir_whitespace_cnt)
+            repl = repl + (" " * (len(repl_str) - len(repl)))
+            contents = f.read().replace(bytes(repl_str, 'ascii'), bytes(repl, 'ascii'))
```

Comment:
Sorry, but this approach is simply not going to cut it. I don't want to perform a broad find/replace on the binary. There are too many potential pitfalls.

There are other, more robust ways that we could come up with, like:
* Putting configuration variables / structure in a dedicated ELF section
* Putting configuration in the Staticx Archive